File: clients/android/app/src/main/kotlin/com/lextures/android/core/lms/LmsFeatureModelsAdvising.kt
package com.lextures.android.core.lms

import kotlinx.serialization.Serializable

@Serializable
data class AdvisingNote(
    val id: String,
    val studentId: String,
    val advisorId: String,
    val content: String,
    val visibleToStudent: Boolean,
    val createdAt: String,
    val advisorEmail: String? = null,
    val advisorDisplayName: String? = null,
)

@Serializable
data class AdvisingNotesResponse(
    val notes: List<AdvisingNote>? = null,
)

@Serializable
data class AdvisingRequirementGroup(
    val group: String,
    val coursesRemaining: Int,
)

@Serializable
data class DegreeProgress(
    val configured: Boolean = false,
    val completionPercent: Int? = null,
    val remainingRequiredCount: Int? = null,
    val remainingRequirements: List<AdvisingRequirementGroup>? = null,
    val atRisk: Boolean? = null,
    val lastUpdated: String? = null,
    val stale: Boolean? = null,
    val appointmentUrl: String? = null,
    val recentNotesCount: Int? = null,
)

@Serializable
data class MyAdvisingConfig(
    val appointmentUrl: String? = null,
)

data class AdvisingAdvisorInfo(
    val displayName: String,
    val email: String? = null,
)

File: clients/ios/Lextures/Core/LMS/LMSFeatureModelsAdvising.swift
import Foundation

// MARK: - Academic advising (M7.8)

struct AdvisingNote: Codable, Identifiable, Equatable {
    let id: String
    let studentId: String
    let advisorId: String
    let content: String
    let visibleToStudent: Bool
    let createdAt: String
    let advisorEmail: String?
    let advisorDisplayName: String?
}

struct AdvisingNotesResponse: Decodable {
    let notes: [AdvisingNote]?
}

struct AdvisingRequirementGroup: Codable, Equatable {
    let group: String
    let coursesRemaining: Int
}

struct DegreeProgress: Codable, Equatable {
    let configured: Bool
    let completionPercent: Int?
    let remainingRequiredCount: Int?
    let remainingRequirements: [AdvisingRequirementGroup]?
    let atRisk: Bool?
    let lastUpdated: String?
    let stale: Bool?
    let appointmentUrl: String?
    let recentNotesCount: Int?
}

struct MyAdvisingConfig: Decodable, Equatable {
    let appointmentUrl: String?
}

struct AdvisingAdvisorInfo: Equatable {
    let displayName: String
    let email: String?
}
